src/api/products/use-products.ts

```diff
@@ -0,0 +1,19 @@
+import type { AxiosError } from 'axios';
+import { createQuery } from 'react-query-kit';
+
+import { client } from '../common';
+import type { FetchProductsResponse } from './types';
+
+type Response = FetchProductsResponse;
+
+export const useProducts = createQuery<
+  Response,
+  { page: number; items: number },
+  AxiosError
+>({
+  queryKey: ['products'],
```

Comment:
your page should be part of your query key, so that when you fetch a new page it's already at your cache

Reply:
The library seems to automatically add the variables as queryKey.

Reply:
Which library do you mean? I saw that you added the variables to your query key either way but just to make sure we're talking about the same library, here is the docs: https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key